src/main/java/com/iota/iri/service/garbagecollector/GarbageCollector.java

```diff
@@ -0,0 +1,333 @@
+     * process the found jobs.
+     */
+    public GarbageCollector(Tangle tangle, SnapshotManager snapshotManager, TipsViewModel tipsViewModel) {
+        this.tangle = tangle;
+        this.snapshotManager = snapshotManager;
+        this.tipsViewModel = tipsViewModel;
+
+        MilestonePrunerJob.registerInGarbageCollector(this);
+        UnconfirmedSubtanglePrunerJob.registerInGarbageCollector(this);
+    }
+
+    /**
+     * This method tries to restore the previous state of the Garbage Collector by reading the state file that get's
+     * persisted whenever we modify any of the queues.
+     *
+     * It is used to restore the state of the garbage collector between IRI restarts and speed up the pruning
+     * operations. If it fails to restore the state it just continues with an empty state which doesn't cause any
+     * problems with future jobs other than requiring them to perform unnecessary steps and therefore slowing them down
+     * a bit.
+     */
+    public void restoreCleanupJobs() {
+        try {
+            BufferedReader reader = new BufferedReader(
+                new InputStreamReader(
+                    new BufferedInputStream(
+                        new FileInputStream(getStateFile())
+                    )
+                )
+            );
+
+            try {
+                String line;
+                while((line = reader.readLine()) != null) {
+                    String[] parts = line.split(";", 2);
+                    if(parts.length >= 2) {
+                        JobParser jobParser = this.jobParsers.get(parts[0]);
+                        if(jobParser == null) {
+                            throw new GarbageCollectorException("could not determine a parser for cleanup job of type " + parts[0]);
+                        }
+
+                        addJob(jobParser.parse(parts[1]));
+                    }
+                }
+            } finally {
+                reader.close();
+            }
+        }
+        catch(IOException e) { /* do nothing */ }
+        catch(Exception e) {
+            log.error("could not load local snapshot file", e);
+        }
+    }
+
+    /**
+     * This method adds a job to the GarbageCollector, that will get processed on its next run.
+     *
+     * It first registers the garbage collector in the job, so the job has access to all the properties that are
+     * relevant for its execution, and that do not get passed in via its constructor. Then it retrieves the queue based
+     * on its class and adds it there.
+     *
+     * After adding the job to its corresponding queue we try to consolidate the queue and persist the changes in the
+     * garbage collector state file.
+     *
+     * @param job the job that shall be executed
+     * @throws GarbageCollectorException if anything goes wrong while adding the job
+     */
+    public void addJob(GarbageCollectorJob job) throws GarbageCollectorException {
```

Comment:
This would imply that there would also be a method for just adding without consolidation.

The consolidation is an internal mechanism of the Manager to reduce its memory and disk consumption - as the user of the Manager i don't care about how it performs its task - i just wannt to hand over the tasks and that's it. Maybe it get's a bit clearer if we seperate the interfaces from their implementation.